lib/dispatcher.js

```diff
@@ -4,7 +4,12 @@ Router.prototype.dispatcher = function () {
   var trie = this.trie
 
   return function* trieRouter (next) {
-    var match = trie.match(this.request.path)
+    try {
+      var match = trie.match(this.request.path)
+    } catch (err) {
+      err.code = 1101;
```

Comment:
@jonathanong got it about the readable code... but facing a bit of a problem on the tests-
```
request(server)
.get('/%')
.expect(404, function(err, res) {
  console.log(err == null) // => true
})
```
Does this mean the err is not being properly thrown? Or, is that callback function's `err` argument representing an error from the request itself?
```javascript
try {
  request(server)
  .get('/%')
  .expect(404, done)
} catch(e) {
  console.log(e)
}
```
the `catch` block is also not hit in the above example, but that's kinda expected since I wouldn't expect superagent (`request`) to throw an error there.

any ideas on how to get a hold of the thrown error here so i can assert that code is some value?